Marlin/src/pins/pins_postprocess.h

```diff
@@ -456,6 +456,11 @@
   #define TEMP_BED_PIN -1
 #endif
 
+// Use ATEMP if TEMP_SOC_PIN is not defined
+#if !defined(TEMP_SOC_PIN) && defined(ATEMP)
```

Comment:
ATEMP is defined in STM32 environment and maybe not in others. Isn't better to define TEMP_SOC_PIN directly in each MB pins definition files? I know it's a big task but it seems logic since this may be seen as an internal pin and doing that you may define it or not depending on SoC used by the board itself.

Reply:
>  Isn't better to define `TEMP_SOC_PIN` directly in each MB pins definition files?

If the platform provides the MCU's internal pin number as `ATEMP` then we don't have to define it in individual pins files. Users are encouraged to set custom pins in their configurations rather than edit pins files, so repeating this item in many pins files offers no special benefit for configuration, or indeed even for board documentation. Presumably all boards with certain MCUs will have an `ATEMP` so it's not a board-related feature, _per se_.

Reply:
yep but not all SoCs have the registers to properly read internal temperature, and there is no way (I found) to properly discriminate SoC inside its family. Defining it into pins may be a solution. It will be defined only for boards that uses the compatible SoCs and users may be free to change it as they want (as it's done for all other pins). SAMD51 has 2 sensors then another decision should be taken (ok, it's not supported there but it is just a way to show different platforms behaviour)